Automattic/jetpack
# Search: better module control error reporting

#### Changes proposed in this Pull Request:

Add proper error messages for module activation. More specifically:

- Return `connection_required` error if site is not connected
- Return `site_offline` error if the site is in offline mode
- Use Search `Module_Control` in rest API rather than use Jetpack `Modules` directly for abstraction reasons
- Fixed unit tests where `jetpack_options` is not filtered correctly
- Added tests for the errors

#### Jetpack product discussion
p1653072333653809-jetpack-crew-slack

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Open a JP site with Jetpack Search subscription
* Open `/wp-admin/admin.php?page=jetpack-search` 
* Ensure you could enable/disable search module
* Ensure unit tests pass
